### PR TITLE
Note thunk

### PR DESCRIPTION
**Pull request by tiffanybacher**
Created a thunk to handle adding a new note to server as well as updating Redux state with note, loading, and any errors.

### CODE REVIEW

**Comment by Klhalls89** [comment]
excellent looking thunk, and thanks for adding in those semicolons

File: src/actions/index.js [hunk 1]
```diff
@@ -1,19 +1,29 @@
...
 export const toggleComplete = (bool) => ({
   type: 'TOGGLE_COMPLETE',
   complete: bool
-})
\ No newline at end of file
+});
+
+export const setLoading = (bool) => ({
+  type: 'SET_LOADING',
+  isLoading: bool
```
Comment (letakeane) [inline src/actions/index.js]: Example comment/code review start

File: src/reducers/index.js [hunk 2]
```diff
@@ -1,6 +1,10 @@
 import { combineReducers } from 'redux';
-import { noteListReducer } from './noteReducer'
+import { notesReducer } from './notesReducer';
+import { isLoadingReducer } from './isLoadingReducer';
```
Comment (letakeane) [inline src/reducers/index.js]: More comments

File: src/thunks/fetchAddNote.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      body: JSON.stringify(body)
+    }
+
+    try {
+      dispatch(setLoading(true));
+
+      const response = await fetch(url, options);
+
+      if (!response.ok) {
+        throw Error(response.statusText);
```
Comment (letakeane) [inline src/thunks/fetchAddNote.js]: Question????

File: src/actions/index.js (hunk 1 above)
Comment (letakeane) [reply src/actions/index.js]: More discussion/explanation

**Pull request merged**